envoy/upstream/cluster_manager.h

```diff
@@ -150,7 +150,6 @@ struct ClusterConnectivityState {
   ~ClusterConnectivityState() {
     ASSERT(pending_streams_ == 0);
     ASSERT(active_streams_ == 0);
-    ASSERT(connecting_and_connected_stream_capacity_ == 0);
```

Comment:
I discussed this in the parent PR, but I think we should keep this here. The change in source/common/http/http3/conn_pool.cc, I believe, should make this chance unnecessary.

https://github.com/envoyproxy/envoy/pull/26107#discussion_r1140678979

Reply:
Hm. I'm proposing that this change be reverted and that the following line be restored:

```
    ASSERT(connecting_and_connected_stream_capacity_ == 0);
```

Given that this PR has been ... challenging to land we can add this back in a follow up, if you prefer.